decidim/decidim
# Fix CSS overrides in applications

#### :tophat: What? Why?
When we have switched away from webpacker to Shakapacker back into #10389, we accidentally introduced a bug regarding the CSS overrides. This may not be visible at the first hand, but when you try to create overrides for component css ( meetings, proposals), you are forced to use `!important`. 

As per current implementation, the `decidim_application.scss` is being imported as part of `decidim_core.scss`. However, in the Decidim + Shakapacker world, the component specific css is being required as a standalone file, which means that the css overrides is being loaded before the code that is supposed to override. 

This PR fixes this by adding 3 new entrypoints: `decidim_overrides`, `decidim_admin_overrides` and `decidim_system_overrides`, which are loaded after the other css has been loaded. 

Basically, in the frontend context: 

```
<link rel="stylesheet" media="all" href="/decidim-packs/css/decidim_core.css" />
<link rel="stylesheet" media="all" href="/decidim-packs/css/decidim_meetings.css" />
```

Becomes:
```
<link rel="stylesheet" media="all" href="/decidim-packs/css/decidim_core.css" />
<link rel="stylesheet" media="all" href="/decidim-packs/css/decidim_meetings.css" />
<link rel="stylesheet" media="all" href="/decidim-packs/css/decidim_overrides.css" />
```

This will allow any user to properly customize the front-end, admin or system areas of the platform.

#### :pushpin: Related Issues
*Link your PR to an issue*
- Fixes #12645

#### Testing

1. Start the application
2. Visit the project homepage
3. Inspect the upcoming meetings widget classes and identify meeting-list__block-list class
4. Inspect the applied css, and see the display: property ( it should be display: flex)
5. Edit app/packs/stylesheets/decidim/decidim_application.scss
6. Add an override like .meeting-list__block-list { display: block; }
7. Refresh and repeat steps 3 & 4 , observe the active css display rule is still flex
8. Our override is like 3rd in the applied rules, and is being overwritten
9. Apply patch, and restart the application
10. Repeat steps 3 & 4. Identify the overridden class, observe the display property. 
11. It should be `display: block` 

:hearts: Thank you!
